src/main/java/org/osc/manager/rest/server/api/DomainApis.java

```diff
@@ -80,36 +147,268 @@ public PolicyEntity updatePolicy(@PathParam("domainid") Long domainid, @PathPara
     public PolicyEntity deletePolicy(@PathParam("domainid") Long domainid, @PathParam("policyid") Long policyid,
             PolicyEntity entity) {
 
-        logger.info("Deleting Policies Entity ...");
-        //TODO Sudhir: Add db calls here
-        return null;
+        logger.info("Deleting for Policies Entity Id...:" + policyid);
+
+        return this.txControl.required(new Callable<PolicyEntity>() {
+
+            @Override
+            public PolicyEntity call() throws Exception {
+
+                CriteriaBuilder cb = em.getCriteriaBuilder();
+
+                CriteriaQuery<PolicyEntity> q = cb.createQuery(PolicyEntity.class);
+                Root<PolicyEntity> r = q.from(PolicyEntity.class);
+                q.select(r).where(cb.and(cb.equal(r.get("domain").get("Id"), domainid)),
+                        cb.equal(r.get("id"), policyid));
+
+                List<PolicyEntity> result = DomainApis.this.em.createQuery(q).getResultList();
+                if (result.isEmpty() == false) {
+                    em.remove(result.get(0));
+                    return new PolicyEntity();
```

Comment:
This is strange, deletes should never return anything and if the id is not found it should just succeed (idempotency)

Reply:
ok and made delete as a void return type.